### DISCUSSION TITLE
paddleocr 2.9.1 一运行rec_svtrv2_ch.yml就爆显存 Out of memory error on GPU 0. Cannot allocate 45.000000MB memory on GPU 0, 15.861450GB memory has been allocated and available memory is only 27.125000MB.

### DISCUSSION THREAD

**Discussion by nissansz** [PaddlePaddle/PaddleOCR#14101] [Q&A]
Out of memory error on GPU 0. Cannot allocate 45.000000MB memory on GPU 0, 15.861450GB memory has been allocated and available memory is only 27.125000MB.

--------------------------------------
C++ Traceback (most recent call last):
--------------------------------------
0   layer_norm_ad_func(paddle::experimental::Tensor const&, paddle::optional<paddle::experimental::Tensor> const&, paddle::optional<paddle::experimental::Tensor> const&, float, int, bool)
1   paddle::experimental::layer_norm(paddle::experimental::Tensor const&, paddle::optional<paddle::experimental::Tensor> const&, paddle::optional<paddle::experimental::Tensor> const&, float, int, bool)
2   void phi::LayerNormKernel<float, phi::GPUContext>(phi::GPUContext const&, phi::DenseTensor const&, paddle::optional<phi::DenseTensor> const&, paddle::optional<phi::DenseTensor> const&, float, int, bool, phi::DenseTensor*, phi::DenseTensor*, phi::DenseTensor*)
3   float* phi::DeviceContext::Alloc<float>(phi::TensorBase*, unsigned long, bool) const
4   phi::DeviceContext::Impl::Alloc(phi::TensorBase*, phi::Place const&, paddle::experimental::DataType, unsigned long, bool) const
5   phi::DenseTensor::AllocateFrom(phi::Allocator*, paddle::experimental::DataType, unsigned long)
6   paddle::memory::allocation::StatAllocator::AllocateImpl(unsigned long)
7   paddle::memory::allocation::Allocator::Allocate(unsigned long)
8   paddle::memory::allocation::Allocator::Allocate(unsigned long)
9   paddle::memory::allocation::Allocator::Allocate(unsigned long)
10  paddle::memory::allocation::CUDAAllocator::AllocateImpl(unsigned long)
11  std::string phi::enforce::GetCompleteTraceBackString<std::string >(std::string&&, char const*, int)
12  phi::enforce::GetCurrentTraceBackString[abi:cxx11](bool)

----------------------
Error Message Summary:
----------------------
ResourceExhaustedError: 

Out of memory error on GPU 0. Cannot allocate 45.000000MB memory on GPU 0, 15.861450GB memory has been allocated and available memory is only 27.125000MB.

Please check whether there is any other process using GPU 0.
1. If yes, please stop them, or start PaddlePaddle on another GPU.
2. If no, please decrease the batch size of your model. 
If the above ways do not solve the out of memory problem, you can try to use CUDA managed memory. The command is `export FLAGS_use_cuda_managed_memory=false`.
 (at /paddle/paddle/fluid/memory/allocation/cuda_allocator.cc:95)

--------------------------------------
C++ Traceback (most recent call last):
--------------------------------------
0   paddle::pybind::ThrowExceptionToPython(std::__exception_ptr::exception_ptr)

----------------------
Error Message Summary:
----------------------
FatalError: `Process abort signal` is detected by the operating system.
  [TimeInfo: *** Aborted at 1729900280 (unix time) try "date -d @1729900280" if you are using GNU date ***]
  [SignalInfo: *** SIGABRT (@0x102) received by PID 258 (TID 0x7d3f2e146740) from PID 258 ***]

**Comment by GreatV** [comment]
batch size 设小点，换大显存的gpu

**Reply by nissansz** [reply]
batch size是1，显存是16G，也是一样的错误。
需要哪个版本的paddlepaddle? 训练rec_svtrv2_ch.yml

**Reply by GreatV** [reply]
换个电脑试试

**Reply by nissansz** [reply]
需要什么系统，什么版本的paddlepaddle?

**Reply by GreatV** [reply]
https://www.paddlepaddle.org.cn/install/quick

**Reply by nissansz** [reply]
paddlepaddle 2.4.2能不能用？

**Reply by nissansz** [reply]
配置文件里有没有哪个参数可以减小显存的？
    scales: [[320, 32], [320, 48], [320, 64]] 把这个改小？

**Reply by nissansz** [reply]
paddlepadle 2.4.2下 可以加载一些图片，然后就自动退出了

![image](https://github.com/user-attachments/assets/b1abec7b-8ce2-417d-b2c0-20fee78548fe)

**Reply by nissansz** [reply]
用ResNet34的配置是可以训练的。有没有办法改一些参数用ResNet34的，保留rec_svtrv2_ch.yml的核心架构参数，让训练能运行起来
